### ISSUE TITLE
transformers/run_eval.py broken for batched recog with transformers 4.51.3

### ISSUE THREAD

**Issue by albertz** [huggingface/open_asr_leaderboard#68]
[transformers/run_eval.py](https://github.com/huggingface/open_asr_leaderboard/blob/main/transformers/run_eval.py) seems broken for batched recognition.

When running this code (with CrisperWhisper, but not sure if that's relevant), I get the warning:

> You have passed task=transcribe, but also have set `forced_decoder_ids` to [[1, None], [2, 50360]] which creates a conflict. `forced_decoder_ids` will be ignored in favor of task=transcribe.
> The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input’s `attention_mask` to obtain reliable results.

And I get a lot of hallucinations, and overall WER is high due to that, e.g. on Tedlium: WER: 20.51% (sub: 1.31%, del 1.20%, ins 18.00%)

However, when I set `batch_size=1`, then it seems to be fine, and I get 3.09% WER on Tedlium.

So, it seems like sth is wrong with the batching. The warning probably gives a good hint about the missing attention mask?

(I also wonder about the warning about `forced_decoder_ids`, but maybe that's not so important?)

This is with transformers 4.51.3. I assume it has worked before in some earlier transformers version? Which transformers version is used to generate the numbers on https://huggingface.co/spaces/hf-audio/open_asr_leaderboard?

**Comment by albertz** [comment]
Ok, it seems these are related:
* https://github.com/huggingface/transformers/issues/33498
* https://github.com/huggingface/transformers/issues/35524
* https://github.com/huggingface/transformers/issues/32228

**Comment by albertz** [comment]
Adding `return_attention_mask=True` to the `processor` args gets rid of the warning about the attention mask, but it doesn't seem to resolve the problem. I get exactly the same as before (20.51% WER).
